### PR TITLE
Fix okhttp implementation

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Fixes the OkHttp implementation to decode the response using gzip & updates the tests.

## Testing

Tested against the live endpoint in the example app.

### CODE REVIEW

File: embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/config/OkHttpRemoteConfigSource.kt [hunk 1]
```diff
@@ -41,8 +43,11 @@ internal class OkHttpRemoteConfigSource(
         if (!response.isSuccessful) {
             return null
         }
-        val config = response.body?.source()?.inputStream()?.buffered()?.use {
-            serializer.fromJson(it, RemoteConfig::class.java)
+        val config = response.body?.source()?.use { src ->
```
Comment (bidetofevil) [inline embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/config/OkHttpRemoteConfigSource.kt:46]: Is the tiny payload we get back worth gzipping? Or would it be annoying and not worthwhile do do something different here?

**Review by bidetofevil — APPROVED** [review]
LGTM. Comment about whether we should get a compressed response at all - probably find the way it is, just wanted to raise it.

**Review by bidetofevil — APPROVED** [review]
LGTM

File: embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/config/OkHttpRemoteConfigSource.kt (hunk 1 above)
Comment (fractalwrench) [reply embrace-android-core/src/main/kotlin/io/embrace/android/embracesdk/internal/config/OkHttpRemoteConfigSource.kt:46]: I'm easy either way - let's discuss internally & update in a separate PR if necessary

**Pull request merged**